sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java

```diff
@@ -335,13 +339,30 @@ public void testFilteredSelfJoin() throws Exception {
         execute("select more.name, less.name, (more.salary - less.salary) from employees as more, employees as less " +
                 "where more.salary > less.salary " +
                 "order by more.salary desc, less.salary desc");
-        assertThat(printedTable(response.rows()), is("" +
-                                                     "Douglas Adams| Trillian| 200.0\n" +
+        assertThat(printedTable(response.rows()), is("Douglas Adams| Trillian| 200.0\n" +
                                                      "Douglas Adams| Ford Perfect| 600.0\n" +
                                                      "Trillian| Ford Perfect| 400.0\n"));
     }
 
     @Test
+    public void testFilteredSelfJoinWithFilterOnBothRelations() {
+        execute("create table test(id long primary key, num long, txt string) with (number_of_replicas=1)");
+        // The same test is run also by JoinSingleNodeIntegrationTest
+        if (internalCluster().numDataNodes() > 1) {
+            ensureGreen();
+        } else {
+            ensureYellow();
```

Comment:
I'm not 100% sure if my thought was correct but I wanted to have different initial routing for the relations so I wanted to have shard replicas. Maybe since I'm unit testing the updateLocations it's not needed... ?